### PR TITLE
sst_importer: duplicate detection skip rest KV when meet delete

### LINKED ISSUE #14961: duplicate detection of lightning supports cleaning up old data by DELETE
### Is your feature request related to a problem? Please describe:

ref https://github.com/pingcap/tidb/issues/44723

### PR DESCRIPTION
**Pull request by lance6716**
### What is changed and how it works?

Issue Number: Close #14961

What's Changed:

```commit-message
```

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
/cc @kennytm 

I can't find enough reviewer 😂

File: src/import/duplicate_detect.rs [hunk 1]
```diff
@@ -490,6 +498,11 @@ mod tests {
         ];
         let snapshot = storage.get_snapshot();
         let detector = DuplicateDetector::new(snapshot, b"0".to_vec(), None, 13, false).unwrap();
+        check_duplicate_data(detector, expected_kvs.clone());
+
+        // in fact lightning will not set min_commit_ts
+        let snapshot = storage.get_snapshot();
+        let detector = DuplicateDetector::new(snapshot, b"0".to_vec(), None, 0, false).unwrap();
```
Comment (tonyxuqqi) [inline src/import/duplicate_detect.rs:505]: I did not understand how this test will test the new logic, as it seems no WriteType::Delete in test case.

**Review by tabokie — APPROVED** [review]

**Review by overvenus — COMMENTED** [review]
Could you add a test?

**Comment by lance6716** [comment]
> duplicate_detect.rs

It's in duplicate_detect.rs line 503. Without the change of code part it will fail, because it won't stop when see delete then more duplicate entry will be returned

**Review by overvenus — APPROVED** [review]

**Pull request merged**